packages/@aws-cdk/aws-stepfunctions-tasks/README.md

```diff
@@ -133,6 +133,29 @@ const submitJob = new tasks.LambdaInvoke(this, 'Invoke Handler', {
 });
 ```
 
+### ResultSelector
+
+You can use [`ResultSelector`](https://docs.aws.amazon.com/step-functions/latest/dg/input-output-inputpath-params.html#input-output-resultselector)
+to reshape the raw result of a Task, Map or Parallel state before it is passed
+to [`ResultPath`](###ResultPath). For service integrations, the raw result contains
+metadata in addition to the response payload. You can use ResultSelector to pick
+fields from the raw result and combine them with static values or fields from the
+context object.
+
+The following example extracts the output payload of a Lambda function Task and combines
+it with some static values.
```

Comment:
> and combines it with some static values.

Was meant to say it's not just for getting the payload only. But I can expand on this if not clear enough.

---

re: `payloadResponseOnly` for `LambdaInvoke` (probably out of scope for this PR): 

Should we consider making this use ResultSelector instead? Probably not since there are some other implications besides the output like the service integration patterns as you noted. The execution history will look different too.

Reply:
got it, that makes sense.

> re: payloadResponseOnly for LambdaInvoke (probably out of scope for this PR):
> Should we consider making this use ResultSelector instead? Probably not since there are some other implications > besides the output like the service integration patterns as you noted. The execution history will look different too.

I had started on making that change within the scope of this PR initially. My first cut fell back to using the service integration pattern with result selector but I paused on it. I think it makes sense to hold off on it unless there's a more compelling reason to make that change.

Also, as this is now a stable module, a change in behaviour for `LambdaInvoke` would be considered breaking

Reply:
> Also, as this is now a stable module, a change in behaviour for LambdaInvoke would be considered breaking

Agreed.

Reply:
Tried to clarify things a bit more, but one of the main motivations for ResultSelector was to handle the service integration metadata.